apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx

```diff
@@ -133,22 +133,58 @@ const getNativeElementsList = (elements: SBEnumType['value']): JSX.Element => {
   );
 };
 
-const RenderArgsTable = ({ hideArgsTable, primaryStory }: { primaryStory: PrimaryStory; hideArgsTable: boolean }) => {
+const slotRegex = /as\?:\s*"([^"]+)"/;
+function withSlotEnhancer(story: PreparedStory) {
+  type InternalComponentApi = {
+    __docgenInfo: { props?: Record<string, { type: { name: string } }> };
+    [k: string]: unknown;
+  };
+  const component = story.component as InternalComponentApi;
+  const docGenProps = component?.__docgenInfo?.props;
+
+  if (!docGenProps) {
+    return component;
+  }
+
+  Object.entries(docGenProps).forEach(([key, argType]) => {
+    const value: string = argType?.type?.name;
+    const match = value.match(slotRegex);
+
+    if (match) {
+      component.__docgenInfo.props![key].type.name = `Slot<\"${match[1]}\">`;
+    }
+  });
+
+  return component;
+}
+
+const RenderArgsTable = ({
+  hideArgsTable,
+  story,
+  argTypes,
+}: {
+  story: PrimaryStory;
+  hideArgsTable: boolean;
+  argTypes: StrictArgTypes;
+}) => {
   const styles = useStyles();
+
+  const { component } = withSlotEnhancer(story);
```

Comment:
Task:

- [ ] verify that this doesn't mutate story object as we don't wanna do that to prevent any unwanted side effects